tools/integration_tests/util/setup/setup.go

```diff
@@ -262,17 +287,40 @@ func LogAndExit(s string) {
 	os.Exit(1)
 }
 
-// Clean the mounted directory
-func CleanMntDir() {
-	dir, err := os.ReadDir(mntDir)
+// CleanUpDir cleans up given directory.
+func CleanUpDir(directory string) {
+	dir, err := os.ReadDir(directory)
 	if err != nil {
 		log.Printf("Error in reading directory: %v", err)
 	}
 
 	for _, d := range dir {
-		err := os.RemoveAll(path.Join([]string{mntDir, d.Name()}...))
+		err := os.RemoveAll(path.Join([]string{directory, d.Name()}...))
 		if err != nil {
 			log.Printf("Error in removing directory: %v", err)
 		}
 	}
 }
+
+// Clean the mounted directory
+func CleanMntDir() {
+	CleanUpDir(mntDir)
+}
+
+func SetupTestDirectory(testDirName string) string {
+	testDirPath := path.Join(MntDir(), testDirName)
+	err := os.Mkdir(testDirPath, DirPermission_0755)
+	if err != nil && !strings.Contains(err.Error(), "file exists") {
+		log.Printf("Error while setting up directory %s for testing: %v", testDirPath, err)
+	}
+	CleanUpDir(testDirPath)
+	return testDirPath
+}
+
+func CleanupTestDirectoryOnGCS(directoryName string) {
```

Comment:
Can we generalize it to gcs directory path ?

Reply:
By generalise, do you mean sending the full object name in parameter instead of 1st level directory name?

Reply:
Yes. 
e.g. testbucket/dir1/dir2 should also remove